controllers/openstackdeploy_controller.go

```diff
@@ -106,6 +107,12 @@ func (r *OpenStackDeployReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		)
 	}
 
+	uuid, _ := exec.Command("uuidgen").Output()
+
+	if instance.Spec.DeployIndentifier == "" {
+		instance.Spec.DeployIndentifier = string(uuid)
+	}
+
```

Comment:
Is something missing? This patch is just setting DeployIdentifier, but it's not used anywhere.

Reply:
If it's changed by the user (spec changes) won't the reconciler run the job (which is in intent)? This is not expected to be generated all the time i.e only when it's not provided by the user.

Reply:
Oh, I thought the intent was to pass this through to tripleo to ensure TRIPLEO_DEPLOY_IDENTIFIER is unique when running containers.

I don't see why it's necessary to add this to the deploy CR. If the user wants to interrupt and rerun the deploy job they could just delete the job resource.

Reply:
Re [1], yes, [2] the config version is significant (or at least it should be, changing doesn't currently work).
I'm not saying that adding this param would not work. However adding a dummy param that has no purpose except to trigger a reconcile seems like an anti-pattern to me. @dprince WDYT?

Reply:
Just to be clear, there are jobs resources (not deploy resources):
```
oc -n openstack get job
NAME                                                              COMPLETIONS   DURATION   AGE
0ea8b7e1b712266856512cc28b8444b1cdfbf30b09aa956c9bdbcbf374f730b   1/1           10s        10h
deploy-openstack-default                                          1/1           35m        3d10h
```

If you want to force a redeployment then I think deleting the deploy-openstack-default job would achieve this. There is no need to add this dummy param to the deploy resource.

Reply:
Or perhaps we should remove patch & update from the deploy CRD and force a new deploy CR to be created for each overcloud deploy run.

Reply:
that would make sense, especially if tracked as git revisions for future devops story

Reply:
My understanding is OpenStackDeploy is a CR which manages lifecycle of 'overcloud configuration' with k8s api. The fact that it uses/creates a Job resource is something internal to it. From the API point of view I would have probably created a CR that mimics overcloud configuration (after osvms and osbms provisioned) including config generation and manages all the things (deploy, re-deloy, update, upgrade) we do with a deployment(?).

Reply:
> My understanding is OpenStackDeploy is a CR which manages lifecycle of 'overcloud configuration'

@dprince? (created this CR)

I think of it as a CR that represents a (single?) invocation of `openstack overcloud deploy <args>`. 

I expect a CR to manage the life-cycle of the overcloud would need to be a level above this, creating deploy CRs for initial deploy, scale-up, minor update, etc...

Reply:
>  There is nothing currently that prevents you from creating another OpenStackDeploy CR to re-run the same playbooks and we could certainly go that route to pass.

Won't that lead to someone creating multiple CRs that can overwrite each others configurations (possibly using different config hash)(?).

Yeah, probably we can try wrap these with higher level mutable CRs for better abstraction.